dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/Relation.java

```diff
@@ -114,6 +121,16 @@ public boolean getIsNamed(){
         return name != null;
     }
 
+    @Override
+    public Collection<? extends Relation> getConstrainedRelations() {
+        return accept(new Visitor<>() {
+            @Override
+            public Collection<? extends Relation> visitDefinition(Relation rel, List<? extends Relation> dependencies) {
+                return dependencies;
+            }
+        });
+    }
+
```

Comment:
I kinda dislike the use of a visitor here.
In general, I think the constrained relations can be assumed to be ordered and hence should be representable by a List<Relation>.
Then, for definitions we could simply maintain the invariant that the first or last entry is the defined relation and the other one's are the dependencies.
This simplifies the handling IMO.

Reply:
The visitor always gets all constrained relations.  There is not need for an extra override-tree.  But the list could be helpful for other users.

Reply:
I understand that and for now this might be ok.
But I'm thinking that eventually all definitions will have an explicit list of their constrained relations (for base definitions, it might be a singleton) which they simply can return.
Either way, for now this is ok.